src/app/settings/base-stix.component.ts

```diff
@@ -249,4 +250,76 @@ export class BaseStixComponent {
         );
     }
 
+    
+    /**
+     * @param  {any} model (Any legacy STIX model)
+     * @returns boolean
+     * @description This is used to disable the save button in the CRUD pages
+     */
+    public invalidate(model: any): boolean {
+        this.validationErrorMessages = [];
+        let invalid = false;
+
+        // created_by_ref not on identity
+        if (model.type !== 'identity' && (!model.attributes.created_by_ref || !model.attributes.created_by_ref.length)) {
+            this.validationErrorMessages.push('Submitter Organization is required');
+            invalid = true;
+        }
+        
+        // Name not on sighting or relationship
+        if (model.type !== 'sighting' && model.type !== 'relationship' && (!model.attributes.name || !model.attributes.name.length)) {
+            this.validationErrorMessages.push('Name is required');
+            invalid = true;
+        }
+        
+        switch (model.type) {
+            case 'identity': 
+                if (!model.attributes.identity_class || !model.attributes.identity_class.length) {
+                    this.validationErrorMessages.push('Identity class is required');
+                    invalid = true;
+                }
+                break;
+            case 'indicator':
+                if (!model.attributes.pattern || !model.attributes.pattern.length) {
+                    this.validationErrorMessages.push('Pattern is required');
+                    invalid = true;
+                }
+                break;
+            case 'report':
+                if (!model.attributes.published) {
+                    this.validationErrorMessages.push('Published is required');
+                    invalid = true;
+                }
+                break;
+            case 'sighting':
+                if (!model.attributes.sighting_of_ref || !model.attributes.sighting_of_ref.length) {
+                    this.validationErrorMessages.push('Sighting reference is required');
+                    invalid = true;
+                }
+                break;
+        }
+
+        // external references
+        if (model.attributes.external_references && model.attributes.external_references.length) {
+            for (let externalReference of model.attributes.external_references) {
+                if (!externalReference.source_name || !externalReference.source_name.length ) {
+                    this.validationErrorMessages.push('All external references must have a source name: Add one, or remove the external reference');
+                    invalid = true;
+                }
+            }
+        }
+
```

Comment:
Only gripe I have for external_references and kill_chain_phases below, is that if I hit the add button several times (say, because I know I will have to add three of them), I get 3 error messages that all say the same thing. Better if I just saw the complaint once.